pkg/local_object_storage/shard/gc.go

```diff
@@ -214,9 +216,12 @@ func (s *Shard) removeGarbage() {
 
```

Comment:
Should we unlock here too?

Reply:
for sure. rechecked all the places

Reply:
Can we move all code to a separate function and use `defer` instead? It is ok to have 1-2 different `Unlock`, we have 4.

Reply:
You can split it into `delete` (no mutex) and `Delete` (take mutex + do `delete`).

Reply:
@fyrchik, agree, fixed. unfortunately, i do not know how to solve that easily for our callbacks that are `engine` level config in fact. do you have any ideas?